src/main/java/frc/robot/vision/sources/RobotPoseEstimatingVisionSource.java

```diff
@@ -0,0 +1,17 @@
+package frc.robot.vision.sources;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.vision.GyroAngleValues;
+import frc.robot.vision.RawVisionPoseData;
+
+import java.util.Optional;
+
+public interface RobotPoseEstimatingVisionSource extends VisionSource {
+
+	void updateGyroAngles(GyroAngleValues gyroAngleValues);
+
+	Optional<Rotation2d> getRobotHeading();
+
+	Optional<RawVisionPoseData> getPoseEstimation();
```

Comment:
```suggestion
	Optional<RawVisionPoseData> getRawVisionRobotPoseData();
```

Reply:
[May be impossible](https://github.com/GreenBlitz/GB-Robot-Template/pull/470#discussion_r1867712837) but meanwhile I changed it to `getRawVisionData`.